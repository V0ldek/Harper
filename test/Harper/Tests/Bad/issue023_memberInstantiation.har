value T a = {
    foo :: a -> a;
    foo n = n;
};

main :: String; 
main = {
    (t :: T Integer) = val T {};
    return t.foo "Str";
};