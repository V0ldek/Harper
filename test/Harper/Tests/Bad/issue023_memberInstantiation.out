Error: expression `"Str"` has invalid type `String`; expected `Integer`.
During evaluation of:
 t . foo "Str"

Located at line 9 column 12


Execution terminated with an error: type error.